core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -138,6 +169,22 @@ public boolean accept(BytesRef value) {
 
     }
 
+    class PartitionedOrdinalsFilter extends OrdinalsFilter {
+        @Override
+        public LongBitSet acceptedGlobalOrdinals(RandomAccessOrds globalOrdinals) throws IOException {
+            final long numOrds=globalOrdinals.getValueCount();
+
+            LongBitSet acceptedGlobalOrdinals = new LongBitSet(numOrds);
+            final int zeroBasedPartition = incPartition-1;
+            for (int i = 0; i < numOrds; i++) {
+                if (i % incNumPartitions == zeroBasedPartition){                
+                    acceptedGlobalOrdinals.set(i);
+                }
+            }
```

Comment:
This is true. The alternative way I was thinking about would be to pull a terms enum from the sorted set doc values, and set bits in the bit set based on the hash of the values rather than their ordinal. This way, the partitioning would be based on the value but the terms aggregation would still be able to leverage ordinals to do the bucketing. The drawback is that it requires to compute a hash on every term of the field.